File: js210/js210_small_problems/easy_1/short_long_short.js
function shortLongShort(word1, word2) {
  let length1 = word1.length;
  let length2 = word2.length;

  if (length1 < length2) {
    return word1 + word2 + word1;
  } else {
    return word2 + word1 + word2;
  }
}

console.log(shortLongShort('abc', 'defgh') === "abcdefghabc");
console.log(shortLongShort('abcde', 'fgh') === "fghabcdefgh");
console.log(shortLongShort('', 'xyz') === "xyz");

File: rb109/Practice/Easy 4/short_long_short.rb
# short_long_short.rb

def short_long_short(str1, str2)
  ordered = [str1, str2].sort { |elem1, elem2| elem1.length <=> elem2.length }
  ordered[0] + ordered[1] + ordered[0]
end

puts short_long_short('abc', 'defgh') == "abcdefghabc"
puts short_long_short('abcde', 'fgh') == "fghabcdefgh"
puts short_long_short('', 'xyz') == "xyz"
